applications/concept-catalogue/src/main/kotlin/no/begrepskatalog/conceptregistration/HarvestEndpoint.kt

```diff
@@ -52,45 +54,62 @@ class HarvestEndpoint(val sqlStore: SqlStore) : CollectionsApi {
     fun appendBegrepToCollection(begrep: Begrep, collectionBuilder: CollectionBuilder): Resource {
         var sourceItself = if (begrep.kildebeskrivelse != null && begrep.kildebeskrivelse.kilde != null && begrep.kildebeskrivelse.kilde.size > 0) begrep.kildebeskrivelse.kilde[0].uri + begrep.kildebeskrivelse.kilde[0].tekst else ""
         val urlForAccessingThisBegrepsRegistration = baseURL + begrep.ansvarligVirksomhet.id + "/" + begrep.id
-        var builder = collectionBuilder.conceptBuilder(urlForAccessingThisBegrepsRegistration)
-                .publisher(begrep.ansvarligVirksomhet.id)
-                .definitionBuilder()
-                    .text(begrep.definisjon, "nb")
-                    .sourcedescriptionBuilder()
-                        .sourceBuilder()
-                            .label(sourceItself, "nb")
-                            .seeAlso(begrep.kildebeskrivelse?.forholdTilKilde?.value ?: "")
-                            .build()
-                        .build()
-                    .audience(AudienceType.Audience.Public)
-                    .scopeNote(begrep.merknad ?: "", "nb")
-                    .scopeBuilder()
-                        .label(begrep.omfang?.tekst ?: "", "nb")
-                        .seeAlso(begrep.omfang?.uri)
-                        .build()
-                    .modified(begrep.gyldigFom)
-                    .build()
-                .identifier(begrep.id)
-                .prefLabelBuilder()
-                    .label(begrep.anbefaltTerm ?: "", "no")
-                    .build()
-                .example(begrep.eksempel, "nb")
-                .subject(begrep.fagområde, "nb")
-                .contactPointBuilder()
-                    .email(begrep.kontaktpunkt?.harEpost ?: "")
-                    .telephone(begrep.kontaktpunkt?.harTelefon ?: "")
-                    .build()
-
-        begrep.bruksområde.forEach { bruksområde -> builder = builder.domainOfUse(bruksområde, "nb") }
-
-        var altLabelBuilder = builder.altLabelBuilder()
+
+        var conceptBuilder = collectionBuilder.conceptBuilder(urlForAccessingThisBegrepsRegistration)
+        val definitionBuilder = conceptBuilder.definitionBuilder()
+        val sourceDescriptionBuilder = definitionBuilder.sourcedescriptionBuilder()
+        val sourceBuilder = sourceDescriptionBuilder.sourceBuilder()
+
+
+        when (begrep.kildebeskrivelse.forholdTilKilde) {
+            Kildebeskrivelse.ForholdTilKildeEnum.EGENDEFINERT -> sourceDescriptionBuilder.sourcetype(SourceType.Source.Userdefined)
+            Kildebeskrivelse.ForholdTilKildeEnum.BASERTPAAKILDE -> sourceDescriptionBuilder.sourcetype(SourceType.Source.BasedOn)
+            Kildebeskrivelse.ForholdTilKildeEnum.SITATFRAKILDE-> sourceDescriptionBuilder.sourcetype(SourceType.Source.QuoteFrom)
+        }
+
+        begrep.kildebeskrivelse.kilde.forEach{
```

Comment:
Will this still work if kildebeskrivelse == null?

Reply:
No, it will blow up, spectacularly!